checkov/sca_package/scanner.py

```diff
@@ -98,3 +100,16 @@ def parse_api_result(self, origin_file_path: Path, response: str) -> dict:
         raw_result = json.loads(decompress_file_gzip_base64(response))
         raw_result['repository'] = str(origin_file_path)
         return raw_result
+
+    def request_wrapper(self, method: str, url: str, headers: Any, data: Optional[Any] = None):
+        # using of "retry" mechanism for 'requests.request' due to unpredictable 'ConnectionError' that appears
+        # from time to time ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer')).
+        remaining_tries = CHECKOV_REQUEST_MAX_TRIES
+        try:
+            remaining_tries -= 1
+            return requests.request(method, url, headers=headers, data=data)
+        except requests.exceptions.ConnectionError as err:
+            if remaining_tries == 0:
+                raise err
+            else:
+                time.sleep(CHECKOV_REQUEST_SLEEP_BETWEEN_TRIES)
```

Comment:
```suggestion
                time.sleep(CHECKOV_REQUEST_SLEEP_BETWEEN_TRIES * (CHECKOV_REQUEST_MAX_TRIES - remaining_tries))
```
how about something like this. instead of linear, a bit of adaptive backoff 🙂 so it waits 0.1, then 0.2. exponential could be also a good approach.

Reply:
interesting. OK let's do it =]